### PR TITLE
Prac 06 feedback

### PR DESCRIPTION
**Pull request by cheyenneIT22**
@hamzah-opal-n Hii, I am so sorry for mentioning so late my code looks roughly okay please do let me know what I can improve on?

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, fuel=42, name=""):
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.fuel = fuel
+        self._odometer = 277
```
Comment (hamzah-opal-n) [inline prac_06/car.py:13]: I think the fuel and _odometer values were just an example to show the required format for the __str__ method, you weren't supposed to actually use them

File: prac_06/car.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+        self._odometer += distance
+        return distance
+
+    def __str__(self):
+        return f"{self.name}, fuel={self.fuel}, odometer={self._odometer}"
+
+    @property
```
Comment (hamzah-opal-n) [inline prac_06/car.py:37]: Was this part of the required modifications?

File: prac_06/car.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def odometer(self):
+        return self._odometer
```
Comment (hamzah-opal-n) [inline prac_06/car.py:39]: Was this part of the required modifications?

File: prac_06/guitar.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        self.cost = cost
+
+    def __str__(self):
+        return "{} ({}) : ${:,.2f}".format(self.name, self.year, self.cost)
+
+    def get_age(self):
+        return CURRENT_YEAR - self.year
+
+    def is_vintage(self):
+        if self.get_age() >= 50:
```
Comment (hamzah-opal-n) [inline prac_06/guitar.py:22]: Why not define the minimum age to be considered "vintage" as a constant?

File: prac_06/guitar.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            return True
+        else:
+            return False
```
Comment (hamzah-opal-n) [inline prac_06/guitar.py:25]: I think you can handle this condition in a single line of code

File: prac_06/guitar_test.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+from guitar import Guitar
+
+guitar1 = Guitar("Gibson L-5", 1922, 16035.40)
+print(guitar1.get_age())
+print(guitar1.is_vintage())
+print(guitar1)
+guitar2 = Guitar("Slayer OMG", 1971, 2000)
+print(guitar2.get_age())
+print(guitar2.is_vintage())
+print(guitar2)
```
Comment (hamzah-opal-n) [inline prac_06/guitar_test.py:14]: Have you also tested a guitar that's exactly 50 years old?

File: prac_06/guitar.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+"""
+CP1404, Prac_06,
+Guitar class.
+"""
```
Comment (hamzah-opal-n) [inline prac_06/guitar.py:4]: Did you remember to time yourself?

File: prac_06/programming_language.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+"""
+CP1404, Prac_06,
+Class for a programming language.
+"""
```
Comment (hamzah-opal-n) [inline prac_06/programming_language.py:4]: Did you remember to time yourself?

File: prac_06/languages.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+print("{} \n{} \n{}".format(ruby, python, visual_basic))
+
+# Create a list of programming languages
+programming_languages = [ruby, python, visual_basic]
+
+# List dynamic programming languages using a for loop and is_dynamic method
+print("The dynamically typed languages are: ")
+for programming_language in programming_languages:
+    if programming_language.is_dynamic():
```
Comment (hamzah-opal-n) [inline prac_06/languages.py:18]: Why not use list comprehension for this?

File: prac_06/programming_language.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def __str__(self):
+        return "{}, {} Typing, Reflection={}, First appeared in {}".format(self.field, self.typing, self.reflection,
+                                                                           self.year)
+
+    def is_dynamic(self):
+        if self.typing == "Dynamic":
```
Comment (hamzah-opal-n) [inline prac_06/programming_language.py:20]: You can handle this condition in a single line

File: prac_06/used_cars.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    """Demo test code to show how to use car class."""
+    my_car = Car(180)
+    my_car.drive(30)
+    print(f"Car has fuel: {my_car.fuel}")
+    print(my_car)
+    limo = Car(100)
+    limo.add_fuel(20)
+    print(limo.fuel)
+    limo.drive(115)
+    print("Odo =", limo.odometer)
```
Comment (hamzah-opal-n) [inline prac_06/used_cars.py:20]: Is this statement necessary?

**Review by hamzah-opal-n — COMMENTED** [review]
Pretty good. Left you some feedback and pointers for improvement 🐈‍⬛ 🐾

File: prac_06/car.py (hunk 1 above)
Comment (cheyenneIT22) [reply prac_06/car.py:13]: Ah Thank you for clarifying that.

File: prac_06/car.py (hunk 3 above)
Comment (cheyenneIT22) [reply prac_06/car.py:39]: I am not sure i don't think so.

File: prac_06/guitar.py (hunk 7 above)
Comment (cheyenneIT22) [reply prac_06/guitar.py:4]: I completely forgot to do so.

**Pull request merged**